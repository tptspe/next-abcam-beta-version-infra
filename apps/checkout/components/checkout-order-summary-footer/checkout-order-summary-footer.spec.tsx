import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  CheckoutOrderSummaryFooter,
  CheckoutOrderSummaryFooterProps,
} from './checkout-order-summary-footer'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('CheckoutOrderSummaryFooter', () => {
  const props: CheckoutOrderSummaryFooterProps = {
    isLoading: false,
    total: { value: 723.2, currency: 'Â£' },
    placeOrderOnClick: jest.fn(),
    termsAndConditionsOnClick: jest.fn(),
  }
  let wrapper: any

  beforeEach(() => {
    wrapper = shallow(<CheckoutOrderSummaryFooter {...props} />)
  })

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render total header correctly', () => {
    const totalHeader = wrapper.find('.totalHeader')
    expect(totalHeader).toHaveLength(1)
    expect(totalHeader.html()).toContain('Order total')
  })

  it('should render total value correctly', () => {
    const totalValue = wrapper.find('.totalValue')
    expect(totalValue).toHaveLength(1)
    expect(totalValue.html()).toContain(
      `${props.total.currency}${props.total.value}`
    )
  })

  it('should render Place order buuton correctly', () => {
    const placeOrder = wrapper.find('Button').at(0)
    expect(placeOrder.html()).toContain('Place order')
  })

  it('shpuld render termsAndConditions correctly', () => {
    const termsAndConditions = wrapper.find('.termsAndConditions').at(0)
    expect(termsAndConditions.html()).toContain(
      'By placing your order, you acknowledge that you read and agreed to'
    )
    const termsAndConditionsButton = termsAndConditions.find('Button')
    expect(termsAndConditionsButton.html()).toContain(
      'Abcam&#x27;s terms and conditions'
    )
  })
})
