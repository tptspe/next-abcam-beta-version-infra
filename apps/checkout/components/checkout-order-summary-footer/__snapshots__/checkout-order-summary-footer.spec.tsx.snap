// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutOrderSummaryFooter should match the snapshot 1`] = `
<div
  className="CheckoutOrderSummaryFooter"
>
  <div
    className="content"
  >
    <div
      className="orderTotal"
    >
      <div
        className="totalHeader"
      >
        <p>
          Order total
        </p>
      </div>
      <div
        className="totalValue"
      >
        <p>
          Â£
          723.2
        </p>
      </div>
    </div>
    <div
      className="placeOrder"
    >
      <Button
        className="button"
        onClick={[MockFunction]}
        size="large"
        variant="primary"
      >
        Place order
      </Button>
    </div>
    <div
      className="termsAndConditions"
    >
      <p>
        By placing your order, you acknowledge that you read and agreed to
        <Button
          as="a"
          bodyClassName="termsAndConditions link"
          className="termsAndConditions button"
          onClick={[MockFunction]}
          size="small"
          variant="tertiary"
        >
          Abcam's terms and conditions
        </Button>
        .
      </p>
    </div>
  </div>
</div>
`;
